### DISCUSSION TITLE
How to setup doctrine_orm for Symfony 5 Project

### DISCUSSION THREAD

**Discussion by Tquentel** [phpro/grumphp#909] [Q&A]
Hi, 

I'm currently trying to add grumphp to my Symfony 5 app 
But i have some problems, i have an error when the hook is triggered 

The Error : 

doctrine_orm
You are missing a "cli-config.php" or "config/cli-config.php" file in your
project, which is required to get the Doctrine Console working. You can use the
following sample as a template:
```
<?php
use Doctrine\ORM\Tools\Console\ConsoleRunner;
// replace with file to your own project bootstrap
require_once 'bootstrap.php';
// replace with mechanism to retrieve EntityManager in your app
$entityManager = GetEntityManager();
return ConsoleRunner::createHelperSet($entityManager);
To skip commit checks, add -n or --no-verify flag to commit command
```

My grumphp.yml file : 
grumphp:
    tasks:
        doctrine_orm:
            skip_mapping: false
            skip_sync: false
            triggered_by: ['php', 'xml', 'yml']

When i create the right file, another error appear : 
Fatal error: Uncaught Error: Call to undefined function GetEntityManager() in xxxxx/config/cli-config.php:6

I tried and search for a solution but i found nothing 

Does someone had the same issue ? Or can help me to resolve it ? 

Thanks

**Comment by veewee** [comment]
Hello @Tquentel,

That task uses the official `doctrine` command provided by doctrine:
https://www.doctrine-project.org/projects/doctrine-orm/en/2.8/reference/tools.html

I remember fixing this issue by adding a cli-config.php file that bootstraps Symfony and loads the entity manager from the container.
Something along these lines:

```php
<?php

use AppKernel;

require dirname(__DIR__).'/config/bootstrap.php';

$kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
$container = $kernel->getContainer();

return $container->get('doctrine.orm.entity_manager');
```

**Reply by Tquentel** [reply]
Hello @veewee,

Thanks for the answer, your file should work on Symfony 3 or older applications
But unfortunatly, this config file do not work on my app i've modify it
`<?php

use App\Kernel;
use Symfony\Component\Dotenv\Dotenv;
use Symfony\Component\ErrorHandler\Debug;
use Symfony\Component\HttpFoundation\Request;

require dirname(__DIR__).'/vendor/autoload.php';

(new Dotenv())->bootEnv(dirname(__DIR__).'/.env');

if ($_SERVER['APP_DEBUG']) {
    umask(0000);

    Debug::enable();
}

$kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG']);
$request = Request::createFromGlobals();
$response = $kernel->handle($request);
$response->send();
$kernel->terminate($request, $response);

$container = $kernel->getContainer();

return $container->get('doctrine.orm.entity_manager');`

But i have another error 
![Capture d’écran du 2021-06-03 10-35-22](https://user-images.githubusercontent.com/20561169/120614360-6f186280-c457-11eb-82b3-c7fd139eb91b.png)

**Reply by veewee** [reply]
The exception tells you what to do: instead of returning an entity manager, you need to provide an entity manager provider instead of an entity manager. You can choose one of these:
https://github.com/doctrine/orm/tree/2.9.x/lib/Doctrine/ORM/Tools/Console/EntityManagerProvider


There is no need to handle any request. However you might need to boot the kernel first in order for the container to fill up.
